Loop/src/loop.hxx

```diff
@@ -517,6 +521,110 @@ public:
+    for (int rank = dim - 1; rank >= 0; --rank) {
+
+      // Nested loops that determine {ni,nj,nk} components of the normal vector
+      for (int nk = -1; nk <= +1; ++nk) {
+        for (int nj = -1; nj <= +1; ++nj) {
+          for (int ni = -1; ni <= +1; ++ni) {
+            // True when the normal vector is normal to a {face, edge, corner}
+            if ((ni == 0) + (nj == 0) + (nk == 0) == rank) {
+              // True when point is on left/right boundary,
+              // and vector is not parallel to a {face,corner,edge}
+              // In either of the 3 directions
+              if ((ni != 0 && bbox[ni < 0 ? 0 : 1][0]) ||
+                  (nj != 0 && bbox[nj < 0 ? 0 : 1][1]) ||
+                  (nk != 0 && bbox[nk < 0 ? 0 : 1][2])) {
+
+                const vect<int, dim> inormal{ni, nj, nk};  //define normal vector
+
+                vect<int, dim> imin, imax;
+                for (int d = 0; d < dim; ++d) { // Set imin, imax in each direction
+                  switch (inormal[d]) {
+                  case -1: // lower boundary
+                    // shift "upwards" from physical boundary by # of ghostpoints
+                    //                                      // xxx|~~~~~~~~~~~~~~|~~~
+                    imin[d] = all_min[d] + nghostzones[d];  // ~~~ ^~~~~~~~~~~~~~ ~~~
+                    imax[d] = int_min[d] + nghostzones[d];  // ~~~ ~~~^~~~~~~~~~~ ~~~
+                    break;
+                  case 0: // interior
+                    // shrink the interior by the # of ghostpoints on each side
+                    //                                      // ~~~|xxxxxxxxxxxxxx|~~~
+                    imin[d] = int_min[d] + nghostzones[d];  // ~~~ ~~~^~~~~~~~~~~ ~~~
+                    imax[d] = int_max[d] - nghostzones[d];  // ~~~ ~~~~~~~~~~^~~~ ~~~
+                    break;
+                  case +1: // upper boundary
+                    // shift "downwards" from physical boundary by # of ghostpoints
+                    //                                      // ~~~|~~~~~~~~~~~~~~|xxx
+                    imin[d] = int_max[d] - nghostzones[d];  // ~~~ ~~~~~~~~~~^~~~ ~~~
+                    imax[d] = all_max[d] - nghostzones[d];  // ~~~ ~~~~~~~~~~~~~^ ~~~
+                    break;
+                  default:
+                    assert(0);
+                  }
+
+                  using std::min, std::max;
+                  // Adjust loop_box index range according to tile min, max
+                  imin[d] = max(tmin[d], imin[d]);
+                  imax[d] = min(tmax[d], imax[d]);
+                }
+
+                // Provide loop_box with bnd_min and bnd_max shifted my # of ghostpoints
+                // so that p.NI can be assigned correct values on the "radiative" boundary.
+                // (NewRad requires knowledge of the normal vectors to its boundary points)
+                loop_box<CI, CJ, CK, VS, N>
+                  (bnd_min+nghostzones, bnd_max-nghostzones, imin, imax, f);
```

Comment:
Can you run `clang-format` on the source file?

Reply:
Yes, I'll make sure to run it before making commits.

Reply:
something like:

```
clang-format-14 -style file -i Loop/src/loop.hxx
```

where the `file` style makes it use the `.clang-format` file that Cactus brings with it.

You may have to check which `clang-format-NN` verison to use since to have it only reformat your changed lines. They are, unfortunately, not stable across versions (and we do not want a formatting war between different versions). 

Cactus prescribes clang-format-9 but CarpetX has been using a newer version (14 seemed to have worked for me in the past).

`git add -p` is also very helpful to pick only reformatting of your own source code (instead of the whole file).